stable/dotnetcore/src/Kubeless.Core/Invokers/CompiledFunctionInvoker.cs

```diff
@@ -1,127 +1,71 @@
+            foreach (var path in dependencies) {
+                var dllName = Path.GetFileName(path);
+                LoadAssembly(path);
+            }
         }
 
-        private Type TryGetNamespacedType(Assembly assembly, string moduleName, int namespacesCount)
+        private Assembly LoadAssembly(string path)
         {
-            var @namespace = GetFunctionNamespaceName(assembly);
-            var fullname = $"{@namespace}.{moduleName}";
-
-            var namespacedType = assembly.GetType(fullname);
-
-            if (namespacedType == null)
-                throw new InvalidOperationException($"Your module ({fullname}) was not found in this assembly.");
-
-            return namespacedType;
+            try {
+                return AssemblyLoadContext.Default.LoadFromAssemblyPath(path);
+            } catch (FileLoadException exc) {
+                // We need to check the exception message because 
+                // LoadFromAssemblyPath does not throw a specialized Exception 
+                // when an Assembly has already been loaded
+                if (exc.Message.Contains("Assembly with same name is already loaded")) {
+                    var assemblyName = AssemblyName.GetAssemblyName(path).Name;
+                    return AssemblyLoadContext.Default.Assemblies.SingleOrDefault(assembly => assembly.GetName().Name == assemblyName);
+                }
+                throw exc;
+            }
         }
 
-        private string GetFunctionNamespaceName(Assembly assembly)
+        public async Task<object> Execute(Event kubelessEvent, Context kubelessContext)
         {
-            return assembly.GetTypes().Select(t => t.Namespace).First();
-        }
-
-        private Assembly LoadAssemblyFromFile(IFunction function)
-        {
-            var Assemblycontent = File.ReadAllBytes(function.FunctionFile);
-            var assembly = Assembly.Load(Assemblycontent);
-
-            return assembly;
-        }
-
-        private void LoadDependentAssemblies(string referencesPath)
-        {
-            var invocationManager = new CustomReferencesManager();
-            var references = invocationManager.GetReferences(referencesPath);
-
-            foreach (var r in references)
-                Assembly.LoadFrom(r);
-        }
-
-        private object InvokeFunction(object[] parameters, Type type, object instance)
-        {
-            // Execute the function
-            return type.InvokeMember(Function.FunctionHandler,
-                                     BindingFlags.Default | BindingFlags.InvokeMethod,
-                                     null,
-                                     instance,
-                                     parameters);
-        }
-
-        public object Execute(CancellationTokenSource cancellationSource, Event kubelessEvent, Context kubelessContext)
-        {
-            // Instantiates a new function
-            object instance = Activator.CreateInstance(functionType);

```

Comment:
I see you are creating the kubeless functions instance just once, on the constructor. This behavior was introduced on the first implementation (back on dotnetcore 1.0), but this was creating some concurrency problems. When creating the function once, we create a singleton - and this may not be desired by folks who use kubeless, since it will make their functions share state.

Can you please create a function instance per request? (making them transient, instead of singleton)